src/main/java/seedu/address/model/client/Birthday.java

```diff
@@ -53,59 +44,25 @@ public String formattedDate() {
         return birthdayFormatted;
     }
 
-    /**
-     * Returns true if birthday fall between the given period {@code dateKeyword} else returns false.
-     */
-    public boolean isBirthdayInPeriod(DateKeyword dateKeyword) {
-        Birthday startDate;
-        Birthday endDate;
-        LocalDate today = LocalDate.now();
-
-        Birthday upcomingBirthday = new Birthday(
-                LocalDate.of(today.getYear(), date.getMonthValue(), date.getDayOfMonth()));
-
-        switch(dateKeyword) {
-        case TOMORROW:
-            startDate = new Birthday(today.plusDays(1));
-            endDate = new Birthday(today.plusDays(1));
-            break;
-        case THIS_MONTH:
-            startDate = new Birthday(today.withDayOfMonth(1));
-            endDate = new Birthday(today.withDayOfMonth(today.getMonth().length(today.isLeapYear())));
-            break;
-        case THIS_WEEK:
-            startDate = new Birthday(today);
-            endDate = new Birthday(today.plusDays(7));
-            break;
-        default:
-            startDate = null;
-            endDate = null;
-        }
-        requireNonNull(startDate);
-        requireNonNull(endDate);
-
-        return upcomingBirthday.isBeforeDate(endDate) && upcomingBirthday.isAfterDate(startDate);
-
-    }
-
-    private boolean isBeforeDate(Birthday other) {
-        return date.compareTo(other.date) <= 0;
-    }
-
-    private boolean isAfterDate(Birthday other) {
-        return date.compareTo(other.date) >= 0;
-    }
-
     public static boolean isValidBirthday(String test) {
         return test.matches(VALIDATION_REGEX) || test.equals("");
     }
 
     /**
      * Returns true if and only if the given date {@code birthday} is not after today
-     * and not before {@code yearsBefore}.
+     * and not before {@code YEARS_BEFORE}.
      */
     public static boolean isDateInValidPeriod(Birthday birthday) {
-        return !birthday.date.isAfter(LocalDate.now())
-                && !birthday.date.isBefore(LocalDate.now().minusYears(yearsBefore));
+        return isBeforePeriod(birthday, YEARS_BEFORE);
+    }
+
+    /**
+     * Returns a new Birthday with the current year.
+     */
+    public Birthday upcomingBirthday() {
```

Comment:
what if the birthday is on a leap day? will this method have any issue?

Reply:
hmmm yea this will probably have some error... 
Any ideas of how to solve this? 
For those that are born on 2902, how should their birthday be check ah? this one I not too usre

Reply:
i think its faulty to create a new birthday of a new year. is it possible to do this comparison without year?